src/utils/examples.js

```diff
@@ -6,35 +6,64 @@ export class Examples {
         this.examples = [];
         this.model = model;
         this.select_num = select_num;
+        this.embeddings = {};
+    }
+
+    turnsToText(turns) {
+        let messages = '';
+        for (let turn of turns) {
+            if (turn.role === 'user')
+                messages += turn.content.substring(turn.content.indexOf(':')+1).trim() + '\n';
+        }
+        return messages.trim();
+    }
+
+    getWords(text) {
+        return text.replace(/[^a-zA-Z ]/g, '').toLowerCase().split(' ');
+    }
+
+    async getSimilarity(text1, text2) {
+        if (this.model !== null) {
+            let embeddings1 = null;
+            let embeddings2 = null;
+
+            if (this.embeddings[text1])
+                embeddings1 = this.embeddings[text1];
+            else
+                embeddings1 = await this.model.embed(text1);
+
+            if (this.embeddings[text2])
+                embeddings2 = this.embeddings[text2];
+            else
+                embeddings2 = await this.model.embed(text2);
+
+            return cosineSimilarity(embeddings1, embeddings2);
+
+        } else {
+            const words1 = this.getWords(text1);
+            const words2 = this.getWords(text2);
+            const intersection = words1.filter(word => words2.includes(word));
+            return intersection.length / (words1.length + words2.length - intersection.length);
+        }
     }
 
     async load(examples) {
-        this.examples = [];
-        let promises = examples.map(async (example) => {
-            let messages = '';
-            for (let turn of example) {
-                if (turn.role === 'user')
-                    messages += turn.content.substring(turn.content.indexOf(':')+1).trim() + '\n';
+        this.examples = examples;
+        if (this.model !== null) {
+            for (let example of this.examples) {
+                let turn_text = this.turnsToText(example);
+                this.embeddings[turn_text] = await this.model.embed(turn_text);
```

Comment:
Could put a conditional here that checks if turn_text is an empty string

Reply:
we could. but why would it be empty?

Reply:
I guess this line would only be empty if someone modified the profile, but how about lines 33 and 38?

Reply:
yeah potentially. although I see a bigger problem. it seems to always select the same examples

Reply:
also I think its caused by the init message being empty, not the examples